Update docs (#288)
AY2021S1-CS2103T-T10-4/tp: docs/DeveloperGuide.md

The add attendance mechanism is facilitated by `AddAttendanceCommand`. It extends `Command`.

* `AddAttendanceCommand#execute()`: Performs a validity check and adds a student's attendance for a particular week's
  lesson if all validations passed.

The following sequence diagram shows the interactions between the `Model` and `Logic` components during the execution
of an `AddAttendanceCommand` with user input `add-attendance c\1 l\1 s\1`:

![AddAttendanceSequenceDiagram](images/AddAttendanceSequenceDiagram.png)

1. `Logic` uses the `TutorsPetParser` class to parse the user command.

> 1. A new instance of an `AddAttendanceCommand` object would be created by the `AddAttendanceCommandParser` and returns

Reviewer: ```suggestion
1. A new instance of an `AddAttendanceCommand` object would be created by the `AddAttendanceCommandParser` and returned
```

   to `TutorsPetParser`.
1. `TutorsPetParser` encapsulates the `AddAttendanceCommand` object as a `Command` object which is executed by
   the `LogicManager`.
1. The command execution calls `getFilteredStudentList` and `getFilteredModuleClassList` to get the `targetStudent` and
   `targetModuleClass` respectively using indexes from the user input.
1. As seen from the diagram above, `ModuleClassUtil#addAttendanceToModuleClass()` is then called. Execution of that
   method returns a new `ModuleClass` object with the new attendance of the `targetStudent` added.
1. The `targetModuleClass` in the `model` is then updated with the new `ModuleClass` object.

> 1. The results of the command execution is committed using the `commit()` method for `undo/redo` functionalities.

Reviewer: ```suggestion
1. The change resulting from the command's execution is saved using the `commit(...)` method for the `undo`/`redo` feature.
```
